taipy/gui/builder/_element.py

```diff
@@ -93,47 +95,66 @@ def _parse_property(self, key: str, value: t.Any) -> t.Any:
             return value
         if isinstance(value, FunctionType):
             if key.startswith("on_"):
-                if value.__name__.startswith("<"):
-                    return value
-                return value.__name__
-
-            try:
-                source = inspect.findsource(value)
-                st = ast.parse("".join(source[0]))
-                lambda_by_name: t.Dict[str, ast.Lambda] = {}
-                _LambdaByName(self._ELEMENT_NAME, source[1], lambda_by_name).visit(st)
-                lambda_fn = lambda_by_name.get(
-                    key,
-                    lambda_by_name.get(_LambdaByName._DEFAULT_NAME, None) if key == self._DEFAULT_PROPERTY else None,
-                )
-                if lambda_fn is not None:
-                    args = [arg.arg for arg in lambda_fn.args.args]
-                    targets = [
-                        compr.target.id  # type: ignore[attr-defined]
-                        for node in ast.walk(lambda_fn.body)
-                        if isinstance(node, ast.ListComp)
-                        for compr in node.generators
-                    ]
-                    tree = _TransformVarToValue(self.__calling_frame, args + targets + _python_builtins).visit(
-                        lambda_fn
-                    )
-                    ast.fix_missing_locations(tree)
-                    if sys.version_info < (3, 9):  # python 3.8 ast has no unparse
-                        string_fd = io.StringIO()
-                        _Unparser(tree, string_fd)
-                        string_fd.seek(0)
-                        lambda_text = string_fd.read()
-                    else:
-                        lambda_text = ast.unparse(tree)
-                    lambda_name = f"__lambda_{uuid.uuid4().hex}"
-                    self._lambdas[lambda_name] = lambda_text
-                    return f'{{{lambda_name}({", ".join(args)})}}'
-            except Exception as e:
-                _warn("Error in lambda expression", e)
+                return value if value.__name__.startswith("<") else value.__name__
+            # Parse lambda function
+            if (lambda_name := self.__parse_lambda_property(key, value)) is not None:
+                return lambda_name
+        # Embed value in the caller frame
+        if isinstance(value, Decimator) and "decimator" in key:
```

Comment:
this could be `not instance(value, str) and "decimator" in key` => no `Decimator` dependency